Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -627,6 +720,7 @@ void CGPathAddCurveToPoint(CGMutablePathRef path,
     newSink->EndFigure(D2D1_FIGURE_END_OPEN);
     FAIL_FAST_IF_FAILED(newSink->Close());
 
+    path->BeginFigure();
```

Comment:
I suppose I don't understand why a beginfigure is necessary for this; comments  may help!